support/cas-server-support-x509-core/src/main/java/org/apereo/cas/adaptors/x509/authentication/principal/DefaultX509AttributeExtractor.java

```diff
@@ -0,0 +1,71 @@
+package org.apereo.cas.adaptors.x509.authentication.principal;
+
+import org.apereo.cas.util.CollectionUtils;
+import org.apereo.cas.util.LoggingUtils;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Get the default set of attributes from an X509 certificate.
+ * @author Hal Deadman
+ * @since 6.4
+ */
+@Slf4j
+public class DefaultX509AttributeExtractor implements X509AttributeExtractor {
```

Comment:
Please make sure you have a dedicated test class for this extractor, and shoot for 100% test coverage as much as possible.